models/baseModels/Invoice/Invoice.ts

```diff
@@ -553,6 +588,17 @@ export abstract class Invoice extends Transactional {
       },
       dependsOn: ['party'],
     },
+    loyaltyProgram: {
+      formula: async () => {
+        const LoyaltyPointsData = await this.fyo.doc.getDoc(
+          ModelNameEnum.Party,
+          this.party
+        );
```

Comment:
here you are getting a doc from the `Party` Doctype. better to use a suitable variable name that relates to its value.